File: 1 - BackEnd/Vanilla/PHP/estructuras_de_control/match.php
<?php

/* ##########===============########## */
/* ######===--- Try Catch ---===###### */
/* ##########===============########## */

/* La estructura de control (match) funciona de la misma manera que 
la estructura de control (switch).

La diferencia es que (match) retorna el valor que es seleccionado. */

$number = 3;
echo match($number) {
	0 => "Numbero 0",
	1 => "Numbero 1",
	2 => "Numbero 2",
	3 => "Numbero 3",
	4 => "Numbero 4",
	5 => "Numbero 5",
	6 => "Numbero 6",
	7 => "Numbero 7",
	8 => "Numbero 8",
	9 => "Numbero 9",
	10 => "Numbero 10",
	default => "Ningun numero seleccionado."
}

// ---------------------------- //
// ------ Retornar valor ------ //
// ---------------------------- //

/* El valor seleccionado se retorna. */
$usuario = match("1999"){
	"2019" => "Brandon", 
	"2020" => "Anthony", 
	"2022" => "Cosmic",
	default => "Ninguno"
}

# --------------------- #
# ------ Funcion ------ #
# --------------------- #

function hola(string $val = ""){
	return match($val){
		"uno" => "1",
		"dos" => "2",
		"tres" => "3",
		default => "nothing",
	};
}

File: 1 - BackEnd/Vanilla/Python/estructuras-de-control/match.py
# ##########===============##########
# ######===--- Try Catch ---===######
# ##########===============##########

# La estructura de control (match) funciona de manera similar a la estructura de control (switch).
# La diferencia es que (match) retorna el valor que es seleccionado.

number = 3

match number:
    case 0:
        print("Numero 0")
    case 1:
        print("Numero 1")
    case 2:
        print("Numero 2")
    case 3:
        print("Numero 3")
    case 4:
        print("Numero 4")
    case 5:
        print("Numero 5")
    case 6:
        print("Numero 6")
    case 7:
        print("Numero 7")
    case 8:
        print("Numero 8")
    case 9:
        print("Numero 9")
    case 10:
        print("Numero 10")
    case _:
        print("Ningun numero seleccionado.")

# ---------------------------- #
# ------ Retornar valor ------ #
# ---------------------------- #

# El valor seleccionado se retorna.
usuario = match "1999":
    case "2019":
        "Brandon"
    case "2020":
        "Anthony"
    case "2022":
        "Cosmic"
    case _:
        "Ninguno"

# --------------------- #
# ------ Funcion ------ #
# --------------------- #

# Definición de una función usando match
def hola(val: str = ""):
    match val:
        case "uno":
            return "1"
        case "dos":
            return "2"
        case "tres":
            return "3"
        case _:
            return "nothing"

# Ejemplo de uso de la función
print(hola("dos"))  # Imprime: 2
